File: aspnet/web-forms/overview/older-versions-getting-started/master-pages/specifying-the-master-page-programmatically-cs/samples/sample19.cs
protected override void OnPreInit(EventArgs e) 
{ 
	SetMasterPageFile();
	base.OnPreInit(e); 
} 
protected virtual void SetMasterPageFile()
{ 
	this.MasterPageFile = GetMasterPageFileFromSession();
} 
protected string GetMasterPageFileFromSession() 
{ 
	if (Session["MyMasterPage"] == null) 
		return "~/Site.master";
	else
		return Session["MyMasterPage"].ToString(); 
}

File: aspnet/web-forms/overview/older-versions-getting-started/master-pages/specifying-the-master-page-programmatically-vb/samples/sample19.vb
Protected Overrides Sub OnPreInit(ByVal e As System.EventArgs)
 SetMasterPageFile() 
 MyBase.OnPreInit(e)
End Sub 
Protected Overridable Sub SetMasterPageFile() 
 Me.MasterPageFile = GetMasterPageFileFromSession() 
End Sub 
Protected Function GetMasterPageFileFromSession() As String 
 If Session("MyMasterPage") Is Nothing Then
 Return "~/Site.master"
 Else 
 Return Session("MyMasterPage").ToString() 
 End If 
End Function